packages/env/lib/config/parse-config.js

```diff
@@ -114,6 +114,33 @@ function parseSourceString( sourceString, { workDirectoryPath } ) {
 		};
 	}
 
+	// SSH URLs (git)
+	const supportedProtocols = [ 'ssh:', 'git+ssh:' ];
+	try {
+		const sshUrl = new URL( sourceString );
+		if ( supportedProtocols.includes( sshUrl.protocol ) ) {
+			const pathElements = sshUrl.pathname
+				.split( '/' )
+				.filter( ( e ) => !! e );
+			const basename = pathElements
+				.slice( -1 )[ 0 ]
+				.replace( /\.git/, '' );
+			const workingPath = path.resolve(
+				workDirectoryPath,
+				...pathElements.slice( 0, -1 ),
+				basename
+			);
+			return {
+				type: 'git',
+				url: sshUrl.href.split( '#' )[ 0 ],
+				ref: sshUrl.hash.slice( 1 ) || 'master',
```

Comment:
Does this work for repos that have a different main branch (Like Gutenberg)? Or would you pass, e.g. `ssh://git@github.com/WordPress/WordPress.git#trunk`?

Reply:
Yeah, probably something more could be done here, but I copied what was done with the HTTP source (i.e. defaulting to master if the ref isn't provided).